### PR TITLE
BaSP-MR-105: Add redux to put and delete subscription

### PR DESCRIPTION
**Pull request by santimenendez98**
Add redux for subscriptions, modify the form file with all the asynchronous functions in the thunk file, these functions are put in global states.
For the put request, edit the form by accessing the global states for each input, using the get request to access members and classes, pressing submit on the form and getting a success response, it will return a successful modal if everything went well and you it will return to the table, and if the request was wrong it leaves you in the form and shows you an error modal.
And for the delete, use an asynchronous function that will delete the object with the id and return a modal

### CODE REVIEW

**Review by francoax — CHANGES_REQUESTED** [review]
Be carefull, in the list there is a Loop of requests and in the Edit doesn't work if you only change an input and I think is due to you are sending an object and I think you have to send the respective ID of the class or member

**Comment by francoax** [comment]
The delete works !! Nice

File: src/Redux/Subscriptions/thunks.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/api/subscription/${id}`, {
+        method: 'PUT',
+        headers: {
+          'Content-Type': 'application/json'
+        },
+        body: body
+      });
+      const data = await response.json();
+      if (response.status === 200) {
+        const updatedSubscriptions = subscriptions.map((sub) => {
```
Comment (jramire5) [inline src/Redux/Subscriptions/thunks.js]: line 37-45 are needed? , i mean, where are you using the subscription saved on the set?
Can be done on a global variable using redux?

**Review by francoax — APPROVED** [review]

File: src/Redux/rootReducer.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ import membersReducer from './Members/reducers';
 import subscriptionsReducer from './Subscriptions/reducers';
 
 const rootReducers = combineReducers({
-  activities: activititesReducer,
```
Comment (jramire5) [inline src/Redux/rootReducer.js]: delete this change, is crashing activities

File: src/Redux/Subscriptions/thunks.js [hunk 3]
```diff
@@ -2,13 +2,17 @@ import {
...
   getSubscriptionsSuccess,
   getSubscriptionsError,
-  getByIdSubscriptionsPending,
-  getByIdSubscriptionsSuccess,
-  getByIdSubscriptionsError,
   addSubscriptionsPending,
   addSubscriptionsSuccess,
-  addSubscriptionsError
+  addSubscriptionsError,
+  putSubscriptionSuccess,
```
Comment (jramire5) [inline src/Redux/Subscriptions/thunks.js]: change put to edit in order to gain consistency

Comment (jramire5) [reply src/Redux/Subscriptions/thunks.js]: and also because in actions is already changed... Why it's working?

**Review by jramire5 — APPROVED** [review]
Subscriptions is working!

**Review by JulianRiedinger7 — COMMENTED** [review]
Tests passed succesfully!

**Comment by gabiholl** [comment]
Some changes should be done but good job! [https://trello.com/c/nKhH9oe2](url)

**Pull request merged**